modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/selectionFilters.spec.ts

```diff
@@ -219,33 +319,44 @@ test.describe('Filters in Data Set Manager', () => {
 		});
 
 		await test.step('Check that the preselected value is checked', async () => {
-			const picklist = await picklistApiHelpers.getPicklist(picklistName);
-
-			const preselectedValuesMultiSelect = page.locator(
-				'.form-control.form-control-tag-group.input-group'
-			);
-
 			await expect(preselectedValuesMultiSelect).toBeVisible();
 
 			await expect(preselectedValuesMultiSelect).toContainText(
 				picklist.listTypeEntries[0].name
 			);
+		});
 
-			const preselectedValueOption = page.getByRole('option', {
-				name: 'Sample Value',
-			});
+		await test.step('And checking multiple selection type allows to save more preselected values', async () => {
+			await page
+				.getByLabel('Preselected Values')
+				.waitFor();
 
-			await clickAndExpectToBeVisible({
-				target: preselectedValueOption,
-				trigger: preselectedValuesMultiSelect,
-			});
+			await page
+				.getByLabel('Preselected Values')
+				.click();
 
-			const preselectedValueOptionCheckbox =
-				preselectedValueOption.locator(
-					'.custom-control-input.invisible'
-				);
+			await page
+				.getByRole('option', {name: picklist.listTypeEntries[1].name})
+				.waitFor();
 
-			await expect(preselectedValueOptionCheckbox).toBeChecked();
-		});
+			await page
+				.getByRole('option', {name: picklist.listTypeEntries[1].name})
+				.click();
+
+			await expect( page
+				.getByText('Only one value is allowed in "Single" selection mode.') )
+				.toBeVisible();
+
+			await page
+				.getByText('Multiple')
+				.click();
+
+			await expect( page
+				.getByText('Only one value is allowed in "Single" selection mode.') )
+				.not
+				.toBeVisible();
```

Comment:
This is fine I suppose, especially if it was 1 to 1 migrated from Poshi, but it's not really useful for a test. Do we really know that checking multiple selection type allows to save more preselected values? Do we know if single selection behaves as expected? With these steps, we just know that a label is toggled in UI. A valuable to test would try to adding multiple values and assert expected outcomes.

Reply:
Thanks @markocikos, I see your point. The core of the expectation here is the `saveAddFilterModal` below, that acts as a check if we are allowed to save the filter again with more than one preselected value by checking the `Multiple` selection mode.
Do you think a further checking on open the filter and expect the preselected values to be there will be more useful? Or maybe check that we are not allow to save in `Single` selection mode, turn to `Multiple` and check that we can save then?

Reply:
> Do you think a further checking on open the filter and expect the preselected values to be there will be more useful? Or maybe check that we are not allow to save in Single selection mode, turn to Multiple and check that we can save then?

The first one, if I'm understanding you, would be a test on fragment. It would be valuable, but it's a separate test, not necessary in the scope of this task.

The second one is more in the line of what I was thinking. In fact, I tested this and found the actual behavior is bugged. We allow saving multiple values on single selection option, meaning that validation does not work correctly:

https://github.com/user-attachments/assets/7b0835dd-3bce-4314-b713-8a29b502c37e

Reply:
In addition, another bug seems to happen when I choose `API Rest Application` option. Surely this is not expected behavior, I would be able to select a different option if I misclicked. Do we have tickets for these bugs?

https://github.com/user-attachments/assets/91a809a6-08ae-41d6-bb58-6d35ac675617

Reply:
I've removed the test, so we could move forward with this PR and create another ticket to implement the behavior  we want and create a proper test for that